packages/lib/src/project/ProxyAdminProject.ts

```diff
@@ -0,0 +1,53 @@
+import isEmpty from 'lodash.isempty';
+
+import Proxy from '../proxy/Proxy';
+import Logger from '../utils/Logger';
+import ProxyAdmin from '../proxy/ProxyAdmin';
+import BaseSimpleProject from './BaseSimpleProject';
+import { toAddress } from '../utils/Addresses';
+import { ContractInterface } from './AppProject';
+import ContractFactory, { ContractWrapper } from '../artifacts/ContractFactory';
+
+const log: Logger = new Logger('ProxyAdminProject');
+
+export default class ProxyAdminProject extends BaseSimpleProject {
+  public proxyAdmin: ProxyAdmin;
+
+  public static async fetch(name: string = 'main', txParams: any = {}, proxyAdminAddress?: string) {
+    const proxyAdmin = proxyAdminAddress ? await ProxyAdmin.fetch(proxyAdminAddress, txParams) : null;
+    return new this(name, proxyAdmin, txParams);
+  }
+
+  constructor(name, proxyAdmin, txParams) {
+    super(name, txParams);
+    this.proxyAdmin = proxyAdmin;
+  }
+
+  public async createProxy(contractClass, { packageName, contractName, initMethod, initArgs, redeployIfChanged }: ContractInterface = {}): Promise<ContractWrapper> {
```

Comment:
Can't we move these 3 methods (create, upgrade, and changeAdmin) to the parent class? Aren't they shared with SimpleProject?

Reply:
Mmmh these methods in `ProxyAdminProject` interact with `ProxyAdmin`, while the same methods in `SimpleProject` don't. That doesn't mean that we can refactor a bit these functions, as most of the code is indeed repeated. Creating a protected function in the parent class to set up `initCallData`, `implementationAddress` and `initMethod` might be a good idea :slightly_smiling_face:

Reply:
How about `create`? I don't see any difference between the two, if you create the ProxyAdmin in getProxyAdmin if needed.